doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -0,0 +1,157 @@
+# Feature Name
+Deterministic Approach for Interface Link bring-up sequence
+
+# High Level Design Document
+#### Rev 0.2
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Abbreviation](#abbreviation)
+  * [References](#references)
+  * [Problem Definition](#problem-definition)
+  * [Background](#background)
+  * [Objective](#objective)
+  * [Proposal](#proposal)
+  * [Proposed Work-Flows](#proposed-work-flows)
+
+# List of Tables
+  * [Table 1: Definitions](#table-1-definitions)
+  * [Table 2: References](#table-2-references)
+
+# Revision
+| Rev |     Date    |       Author                       | Change Description           |
+|:---:|:-----------:|:----------------------------------:|------------------------------|
+| 0.1 | 08/16/2021  | Shyam Kumar                        | Initial version                       
+| 0.2 | 12/13/2021  | Shyam Kumar,  Jaganathan Anbalagan | Added uses-cases, workflows  
+| 0.3 | 01/19/2022  | Shyam Kumar                        | Addressed review-comments    |
+
+
+# About this Manual
+This is a high-level design document describing the need to have determinstic approach for
+Interface link bring-up sequence and workflows for use-cases around it 
+
+# Abbreviation
+
+# Table 1: Definitions
+| **Term**       | **Definition**                                   |
+| -------------- | ------------------------------------------------ |
+| pmon           | Platform Monitoring Service                      |
+| xcvr           | Transceiver                                      |
+| xcvrd          | Transceiver Daemon                               |
+| CMIS           | Common Management Interface Specification        |
+| gbsyncd        | Gearbox (External PHY) docker container          |
+| DPInit         | Data-Path Initialization                         |
+| QSFP-DD        | QSFP-Double Density (i.e. 400G) optical module   |
+
+# References
+
+# Table 2 References
+
+| **Document**                                            | **Location**  |
+|---------------------------------------------------------|---------------|
+| CMIS v5 | [CMIS5p0.pdf](http://www.qsfp-dd.com/wp-content/uploads/2021/05/CMIS5p0.pdf) |
+
+
+# Problem Definition
+
+1.	Presently in SONiC, there is no synchronization between 400G Datapath Init operation and enabling ASIC (NPU/PHY) Tx which may cause link instability during administrative interface enable “config interface startup Ethernet” configuration and bootup scenarios. 
+      
+    For 400G optics module, the Host (NPU/PHY) needs to provide a valid high-speed Tx input signal at the required signaling rate and encoding type prior to causing a DPSM to exit from DPDeactivated state and to move to DP Init transient state.
```

Comment:
Suggestion: replace "For 400G optics module.." with "For CMIS-compliant active (optical) modules...".

Reply:
Sure, I will change as suggested.